server/src/main/java/org/opensearch/index/IndexingPressureService.java

```diff
@@ -25,11 +25,19 @@ public IndexingPressureService(Settings settings, ClusterService clusterService)
         shardIndexingPressure = new ShardIndexingPressure(settings, clusterService);
     }
 
+    public Releasable markCoordinatingOperationStarted(long bytes, boolean forceExecution) {
+        if (isShardIndexingPressureEnabled() == false) {
+            return shardIndexingPressure.markCoordinatingOperationStarted(bytes, forceExecution);
+        } else {
+            return () -> {};
+        }
+    }
+
     public Releasable markCoordinatingOperationStarted(ShardId shardId, long bytes, boolean forceExecution) {
         if (isShardIndexingPressureEnabled()) {
             return shardIndexingPressure.markCoordinatingOperationStarted(shardId, bytes, forceExecution);
         } else {
-            return shardIndexingPressure.markCoordinatingOperationStarted(bytes, forceExecution);
+            return () -> {};
```

Comment:
Methods look pretty confusing. On one occasion for `isShardIndexingPressureEnabled ` we return empty while in another case we invoke `ShardIndexingPressure `

Reply:
When ```isShardIndexingPressureEnabled``` is enabled, which implies we are doing **Shard level** accounting, then ```shardIndexingPressure.markCoordinatingOperationStarted(shardId, bytes, forceExecution)``` with attribute ``shardId`` is invoked. 

For the other case when,  when ``isShardIndexingPressureEnabled`` is disabled, which implies we are only doing **Node level accounting** (and not Shard level), then ``shardIndexingPressure.markCoordinatingOperationStarted(bytes, forceExecution)`` without attribute ``shardId`` is invoked. 

We could have club these but then, even for Node level accounting caller would need to unnecessarily pass shardId or null at plumbing point. This way caller can chose to call ``markCoordinatingOperationStarted`` based on what attributes it has, without worrying about if ShardIndexingPressure is enabled or disabled.